src/pal/src/thread/process.cpp

```diff
@@ -180,6 +185,30 @@ enum FILETYPE
     FILE_DIR   /*Directory*/
 };
 
+#pragma pack(push,1)
```

Comment:
Can you please add a comment why the pack needs to be set to 1? I have figured it out, but it is not obvious when you look at this structure without having the context of the usage for the semaphore name generation.

Reply:
Even better would be to not depend on the explicit packing control. It is just a factory for subtle bugs.